packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -191,32 +192,63 @@ export class RelationshipsController extends TransportController {
         if (!relationship) {
             throw CoreErrors.general.recordNotFound("Relationship", relationshipId.toString());
         }
+        if (!relationship.cache) {
+            await this.updateCacheOfRelationship(relationship);
+        }
+        if (!relationship.cache) {
+            throw this.newCacheEmptyError(Relationship, relationshipId.toString());
+        }
```

Comment:
Why is it necessary to do this manually in all functions? I'd expect the cache to be updated once a corresponding external events is received.

If this is actually necessary, it should probably happen in a separate function, as this piece of code is repeated quite often. Also, the two identical conditions on the same level look funny to me. Should the second rather be moved inside of the body of the first?

Reply:
this is because the cache is optional, so we need to throw something in case the cache is undefined. I copied this sequence from the completeOperation method - personally I would consider omitting the cache update, but didn't feel like this was worth fighting for.

Reply:
Another entry on the "we need to discuss our refactoring existing coding approach" list

Reply:
You were not happy with me putting `relationship.cache!` either

Reply:
I already updated that with a typeguard, removed much copy pasted code.